cmd/tailscale/cli/up.go

```diff
@@ -585,28 +587,68 @@ func checkForAccidentalSettingReverts(flagSet map[string]bool, curPrefs *ipn.Pre
 		}
 		switch flagName {
 		case "":
-			errs = append(errs, fmt.Errorf("'tailscale up' without --reset requires all preferences with changing values to be explicitly mentioned; this command would change the value of flagless pref %q", prefName))
+			return fmt.Errorf("'tailscale up' without --reset requires all preferences with changing values to be explicitly mentioned; this command would change the value of flagless pref %q", prefName)
 		case "exit-node":
-			if !didExitNodeErr {
-				didExitNodeErr = true
-				errs = append(errs, errors.New("'tailscale up' without --reset requires all preferences with changing values to be explicitly mentioned; --exit-node is not specified but an exit node is currently configured"))
+			if prefName == "ExitNodeIP" {
+				missing = append(missing, fmtFlagValueArg("exit-node", fmtSettingVal(exi)))
 			}
 		default:
-			errs = append(errs, fmt.Errorf("'tailscale up' without --reset requires all preferences with changing values to be explicitly mentioned; --%s is not specified but its default value of %v differs from current value %v",
-				flagName, fmtSettingVal(imi), fmtSettingVal(exi)))
+			missing = append(missing, fmtFlagValueArg(flagName, fmtSettingVal(exi)))
+		}
+	}
+	if len(missing) == 0 {
+		return nil
+	}
+	var sb strings.Builder
+	sb.WriteString("'tailscale up' without --reset requires explicitly mentioning all non-default or changing values:\n\n\t")
```

Comment:
In taking this for a spin locally, I think we could improve this beyond my quick proposal with a few easy moves. Right now, this error prints like this:

```
$ tailscale up --accept-dns --accept-routes
'tailscale up' without --reset requires explicitly mentioning all non-default or changing values:

	tailscale up --accept-dns --accept-routes --advertise-routes=10.10.0.0/16,10.11.0.0/16
exit status 1
$
```

A few notes:
- The newline makes the suggested command kind of lost amongst the "exit status 1" and the warning message feel like a part of the command above it. For such a prominent multi-line error, separating it out with an opening and closing newline, like what we do for auth URLs, helps focus people in the right places. 
- Including the phrase "Error" can also help make it clear that this isn't just some warning, but an impassable error. (Technically inclined folks may be fine with the exit code 1, but it's also easy to gloss over.)
- "without --reset" is a little ambiguous. We can clarify this by explicitly saying "re-run your command with --reset" (arguably we could also describe what --reset does, but I think that's obvious enough)
- We can make it more clear that they have two options: --reset and mention all the flags, by changing the language.

In the end, a little update could end up looking like this.

```
$ tailscale up --accept-dns

Error: changing settings via 'tailscale up' requires mentioning all non-default flags.
To proceed, re-run your command with --reset, or use the command below to keep non-default settings:

  $ tailscale up --accept-dns --accept-routes --advertise-routes=10.10.0.0/16,10.11.0.0/16

exit status 1
```

Reply:
It doesn't print "exit status 1". Is your "tailscale" really running "go run"?

Reply:
Went with this for now:

```
bradfitz@tsdev:~/src/tailscale.com$ tailscale up --accept-dns=false
Error: changing settings via 'tailscale up' requires mentioning all
non-default flags. To proceed, either re-run your command with --reset or
specify use the command below to explicitly mention the current value of
all non-default settings:

        tailscale up --accept-dns=false --hostname=tsdevbrad

bradfitz@tsdev:~/src/tailscale.com$ 
```

Will submit that for now. We can tweak tomorrow when we're both online.

Reply:
Ah, yeah, I swapped it around to picture what it'd be like in the end. I didn't realize that message was only for `go run`. 

I'll commit a small tweak to fix a typo in there ("or specify use the command below" → "or use the command below"), but otherwise that looks good to me. Hits all the main points.

Thanks!